flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerNG.java

```diff
@@ -185,4 +187,28 @@ void deliverOperatorEventToCoordinator(
      */
     CompletableFuture<CoordinationResponse> deliverCoordinationRequestToCoordinator(
             OperatorID operator, CoordinationRequest request) throws FlinkException;
+
+    /**
+     * Read current {@link JobResourceRequirements job resource requirements}.
+     *
+     * @return Current resource requirements.
+     */
+    default JobResourceRequirements requestJobResourceRequirements() {
+        throw new UnsupportedOperationException(
```

Comment:
IIRC, there were plans to implement `GET` for the default scheduler (please correct me if I'm wrong).
If it's out of scope of this PR, should we create a ticket and add a `TODO` here?

Reply:
I had that initially, but it felt bizarre because the scheduler doesn't accept JRR, which could make the user wonder why it is even there.

Ultimately, I'd rather block the feature completely than expose incomplete implementation.

@zentol @rkhachatryan, if you think this should still be added, I'm happy to do that

WDYT?

Reply:
Makes sense :+1: , the best we can do I think is to serve meaningfull HTTP response for both `GET` and `PUT` (not something like 500).

Reply:
Will try to keep this in mind for the PR that adds the rest api handlers.